inc/class-wpseo-meta.php

```diff
@@ -365,9 +365,11 @@ public static function get_meta_field_defs( $tab, $post_type = 'post' ) {
 				/* Adjust the no-index text strings based on the post type. */
 				$post_type_object = get_post_type_object( $post_type );
 
-				$field_defs['meta-robots-noindex']['title']        = sprintf( $field_defs['meta-robots-noindex']['title'], $post_type_object->labels->singular_name );
-				$field_defs['meta-robots-noindex']['options']['0'] = sprintf( $field_defs['meta-robots-noindex']['options']['0'], ( ( WPSEO_Options::get( 'noindex-' . $post_type, false ) === true ) ? $field_defs['meta-robots-noindex']['options']['1'] : $field_defs['meta-robots-noindex']['options']['2'] ), $post_type_object->label );
-				$field_defs['meta-robots-nofollow']['title']       = sprintf( $field_defs['meta-robots-nofollow']['title'], $post_type_object->labels->singular_name );
+				if ( $post_type_object ) {
```

Comment:
Sadly, this if-statement actually breaks the interface in the Post -> Advanced tab as certain strings are dependent on these values to properly replace values.

My suggestion would be to remove this if-statement and replace it with the following on line 364:

```php
if ( $post_type === '' ) {
	return array();
}
```

This way if you every run into a situation that `$post_type` hasn't been set (which should in theory, never happen), you won't display these settings at all.